dotCMS/src/main/java/com/dotcms/cache/DynamicTTLCache.java

```diff
@@ -0,0 +1,97 @@
+package com.dotcms.cache;
+
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
+
+
+/**
+ * This class constructs a cache with dynamic TTLs for each key.
+ * Relies on Caffeine for the underlying cache implementation.
+ * @param <K>
+ * @param <Object>
+ */
+public class DynamicTTLCache<K, Object>  {
+
+    public final long defaultTTLInMillis;
+    private final Cache<K, CacheValue> cache;
+
+    public DynamicTTLCache(long maxCapacity) {
+        this(maxCapacity, Long.MAX_VALUE);
```

Comment:
should be Config

Reply:
Well, there is another constructor you can use to specify a TTL.  This one makes the cache permanent - as if there is no ttl